app/presenters/publication_presenter.rb

```diff
@@ -129,9 +129,46 @@ def video_embed_url
     end
   end
 
+  def promotion_choice
+    choice = promotion_choice_details['choice']
+    if choice.empty?
+      if has_legacy_organ_promotion_details?
+        "organ_donor"
+      else
+        "none"
+      end
+    else
+      choice
+    end
+  end
+
+  def promotion_url
+    url = promotion_choice_details['url']
+    url.empty? ? legacy_organ_promotion_details['organ_donor_registration_url'] : url
+  end
```

Comment:
Well, yeah, that's the question I think.  I'm not sure where would be good.  Maybe we should do more in Publisher so that upstream doesn't need to care, or maybe we should do it in content-api?  Or maybe we just live with it?